docs/01-Using-Fleet/13-Vulnerability-Processing.md

```diff
@@ -83,3 +84,115 @@ The matching occurs server-side to make the processing as fast as possible, but
 For example, when running a development instance of Fleet on an Apple Macbook Pro with 16 cores, matching 200,000 CPEs against the CVE database will take around 10 seconds and consume about 3GBs of RAM. 
 
 The CPU and memory usages are in burst once every hour (or the configured periodicity) on the instance that does the processing. RAM spikes are expected to not exceed the 2GBs.
+
+## Detection pipeline
+
+There are several steps that go into the vulnerability detection process. In this section we'll dive into what they are and how it works.
+
+The process has different parts that are more error-prone than others. Each OS and each application developer and maintainer can (and do) have their own way of defining each part of their app. Some Linux distributions are very strict, but each distribution handles things differently.
+
+The whole pipeline exists to compensate for these differences, and it can be divided in two sections:
+
+1. Collection:
+
+```mermaid
+graph TD;
+    A[Host1 send software list]-->D[Normalization of names, versions, etc]
+    B[Host2 send software list]-->D
+    C[Host3 send software list]-->D
+    D-->E[Storage for later processing]
+```
+
+2. Processing
+
+Processing happens in a loop. The default interval is 1hr.
+
+```mermaid
+graph TD;
+    A{Once an hour}-->B[Normalized software list]
+    B-->C[CPE translation]
+    C-->D[CVE detection]
+    D-->E[False positive removal]
+    E-->A
+```
+
+
+### Ingesting software lists from hosts
+
+The ingestion of software varies per platform. We run a `UNION` of several queries in each:
+
+- [macOS](https://github.com/fleetdm/fleet/blob/main/server/service/osquery_utils/queries.go#L315)
+- [Windows](https://github.com/fleetdm/fleet/blob/main/server/service/osquery_utils/queries.go#L478)
+- [Linux](https://github.com/fleetdm/fleet/blob/main/server/service/osquery_utils/queries.go#L391)
+
+This is the first step into normalizing data across platforms, as we try to get all the same data for all different types of software we detect vulnerabilities on.
+
+Ingestion can be resource hungry, both on the hosts and the Fleet server. A lot of work has gone into reducing the resources needed, and it's still ongoing.
+
+### Translating to CPE
+
+With a somewhat normalized list of software, in order to search CVEs for it, we need to derive a [CPE](https://en.wikipedia.org/wiki/Common_Platform_Enumeration) from the vendor, name, version, and OS.
+
+As described briefly above, we do this by translating the NVD database of CPEs into a [sqlite database that helps Fleet do the lookup of CPEs very quickly](https://github.com/fleetdm/nvd).
```

Comment:
Also, in the NVD repo is where we now pull data from CentOS and generate the information to discard false positives (in the future we plan to rename the repository, because it's not only CPE generation; a name I could think of is "async-vuln-pre" or something like that)

Reply:
vulnerability_assistance_data?